packages/packages.go

```diff
@@ -297,11 +297,63 @@ type Filter struct {
 	Capabilities   []string
 	SpecMin        *semver.Version
 	SpecMax        *semver.Version
+	Discovery      *DiscoveryFilter
 
 	// Deprecated, release tags to be removed.
 	Experimental bool
 }
 
+type DiscoveryFilter struct {
+	Fields DiscoveryFilterFields
+}
+
+func NewDiscoveryFilter(filter string) (*DiscoveryFilter, error) {
+	filterType, args, found := strings.Cut(filter, ":")
+	if !found {
+		return nil, fmt.Errorf("could not parse filter %q", filter)
+	}
+
+	var result DiscoveryFilter
+	switch filterType {
+	case "fields":
+		for _, name := range strings.Split(args, ",") {
+			result.Fields = append(result.Fields, DiscoveryField{
+				Name: name,
+			})
+		}
+	default:
+		return nil, fmt.Errorf("unknown discovery filter %q", filterType)
+	}
+
+	return &result, nil
+}
+
+func (f *DiscoveryFilter) Matches(p *Package) bool {
+	if f == nil {
+		return true
+	}
```

Comment:
Just for checking, would this mean that there is no filter defined in the query parameter?
Or is it there just to avoid raising a panic error ?

Reply:
Yes, this is code safety. This would mean that `Matches` is called with a `nil` value, for example [here](https://github.com/elastic/package-registry/pull/1235/files#diff-20fd7b81f7ff43064db94851e0fc5ad6a9852949b07d72a02d48751be912534fR407).

Reply:
Ok, thanks for the explanation!